src/core/mac/mac.hpp

```diff
@@ -663,6 +663,33 @@ class Mac : public InstanceLocator
      */
     bool IsEnabled(void) const { return mEnabled; }
 
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+    /**
+     * This method appends header IEs to a TX-frame according to its
+     * frame control field and if time sync is enabled.
+     *
+     * @param[in]      aIsTimeSync  A boolean indicates if time sync is being used.
+     * @param[in,out]  aFrame       A reference to the TX-frame to which the IEs will be appended.
+     *
+     * @retval OT_ERROR_NONE    If append header IEs successfully.
+     * @retval OT_ERROR_FAILED  If cannot find header IE position in the frame.
```

Comment:
nit: I generally prefer to avoid generic error like `OT_ERROR_FAILED`. Should this be `OT_ERROR_NOT_FOUND` instead?